src/kafka/kafka_clients.clj

```diff
@@ -0,0 +1,132 @@
+(ns kafka.kafka-clients
+  (:import (org.apache.kafka.clients.producer KafkaProducer ProducerRecord)
+           (org.apache.kafka.common.serialization StringSerializer StringDeserializer)
+           (org.apache.kafka.clients.consumer KafkaConsumer)
+           (java.time Duration)))
+
+(def configs
+  (atom {:producer-config {"value.serializer"  StringSerializer
+                           "key.serializer"    StringSerializer
+                           "client.id"         "m-w1kx3xw0nr"
+                           "bootstrap.servers" "localhost:9092"
+                           "acks"              "all"}
+         :consumer-config {"group.id"           "My-Group"
+                           "key.deserializer"   StringDeserializer
+                           "value.deserializer" StringDeserializer
+                           "client.id"          "m-w1kx3xw0nr"
+                           "bootstrap.servers"  "localhost:9092"}}))
+
+(defn build-producer
+  [configs]
+  (let [producer-config (@configs :producer-config)]
+    (new KafkaProducer producer-config)))
+
+(defn build-consumer
+  [configs]
+  (let [consumer-config (@configs :consumer-config)
+        consumer        (new KafkaConsumer consumer-config)
+        _               (.subscribe consumer ["demo"])]
+    consumer))
+
+(def clients
+  (atom {:producer (build-producer configs)
+         :consumer (build-consumer configs)}))
+
+(defn send-msg!
+  [clients topic value]
+  (let [producer (@clients :producer)
+        record   (new ProducerRecord topic value)]
+    (.send producer record)))
+
+(defn read-msg!
+  [clients]
```

Comment:
named as client, but passing an atom

Reply:
But we should avoid hungarian notation (i.e., clients-atom) right?
Should I pass the consumer client here 
(call this function with `(read-msg! (@clients :consumer))`) ?

Reply:
Do you think we should be passing state around, or data?

Reply:
Should pass data